### DISCUSSION TITLE
No hexagonal items shown on map and the camera map is not rotating

### DISCUSSION THREAD

**Discussion by jaballogian** [visgl/deck.gl#6025] [Q&A] [answered]
Hello, 

Based on my question here https://github.com/visgl/deck.gl/issues/6020, I tried to show data into the hexagonal heatmap view.
Also according to this discussion https://github.com/visgl/react-map-gl/discussions/1537, I tried to rotate the camera map along with a point.

1. I already used ```getPosition``` and ```getElevationValue``` for the ```HexagonLayer``` component but the hexagonal items weren't shown on the map as seen in the screenshot below.
2. I also tried to rotate the camera map using ```transitionDuration```, ```transitionInterpolator```, and ```LinearInterpolator``` but still the camera map wasn't rotating.

![image](https://user-images.githubusercontent.com/24468466/127258434-5d157ff2-51c5-4670-bca9-d8b2f4085792.png)

No error or warning in the browser console.

Sample data:

```javascript
[
    {
        "id": 5226,
        "lat": -6.107405,
        "lng": 106.777575,
        "speed": 9,
    },
    ...
]
```

Code:

```javascript
import React, { useState } from 'react'
import ReactMapGL, { LinearInterpolator } from 'react-map-gl'
import {AmbientLight, PointLight, LightingEffect} from '@deck.gl/core'
import {HexagonLayer} from '@deck.gl/aggregation-layers'
import DeckGL from '@deck.gl/react'
import { rowList } from './formatSampleData'

const MAPBOX_TOKEN = ''

const ambientLight = new AmbientLight({
    color: [255, 255, 255],
    intensity: 1.0
})

const pointLight1 = new PointLight({
    color: [255, 255, 255],
    intensity: 0.8,
    position: [-0.144528, 49.739968, 80000]
})

const pointLight2 = new PointLight({
    color: [255, 255, 255],
    intensity: 0.8,
    position: [-3.807751, 54.104682, 8000]
})

const lightingEffect = new LightingEffect({ambientLight, pointLight1, pointLight2})

const material = {
    ambient: 0.64,
    diffuse: 0.6,
    shininess: 32,
    specularColor: [51, 51, 51]
}

const INITIAL_VIEW_STATE = {
    latitude: -6.2088,
    longitude: 106.8456,
    zoom: 5,
    pitch: 40.5,
    bearing: -27
}

const MAP_STYLE = 'mapbox://styles/mapbox/dark-v9'

export const colorRange = [
  [1, 152, 189],
  [73, 227, 206],
  [216, 254, 181],
  [254, 237, 177],
  [254, 173, 84],
  [209, 55, 78]
]

function getTooltip({object}) {
    if (!object) {
        return null
    }
    const lat = object.position[1]
    const lng = object.position[0]
    const count = object.points.length

    return `\
        latitude: ${Number.isFinite(lat) ? lat.toFixed(6) : ''}
        longitude: ${Number.isFinite(lng) ? lng.toFixed(6) : ''}
        ${count} Accidents`
}

export default function HexagonalHeatmap2({
    radius = 1000,
    upperPercentile = 100,
    coverage = 1
}) {
    const [viewport, setViewport] = useState({
        latitude: -6.2088,
        longitude: 106.8456,
        width: '100vw',
        height: '100vh',
        zoom: 5,
        pitch: 40.5,
        bearing: -27
    })
    
    const layers = [
        new HexagonLayer({
            id: 'heatmap',
            colorRange,
            coverage,
            rowList,
            elevationRange: [0, 3000],
            elevationScale: rowList && rowList.length ? 50 : 0,
            extruded: true,
            getPosition: item => [item['lat'], item['lng']],
            getElevationValue: item => item['speed'],
            pickable: true,
            radius,
            upperPercentile,
            material,
            
            transitions: {
                elevationScale: 3000
            }
        })
    ]

    return (
        <DeckGL
            layers={layers}
            effects={[lightingEffect]}
            initialViewState={INITIAL_VIEW_STATE}
            controller={true}
            // getTooltip={getTooltip}
        >
            <ReactMapGL
                {...viewport}
                mapboxApiAccessToken={MAPBOX_TOKEN}
                mapStyle={MAP_STYLE}
                onViewportChange={viewport => {
                    setViewport(viewport)
                }}
                transitionDuration={10000}
                transitionInterpolator={new LinearInterpolator([{ around: [-6.2088, 106.8456]}])}
            />
        </DeckGL>
    )
}
```

I saw from this documentation https://deck.gl/docs/api-reference/aggregation-layers/hexagon-layer#elevationaggregation that the lat and long is swapped.
So I tried to swap the lat and long for ```getPosition``` value into this
```
getPosition: item => [item['lng'], item['lat']],
```

but still no hexagonal items shown on the map.

What's wrong or missing from the code?

**Comment by Pessimistress** [comment, marked as answer]
- `getElevationValue` is called with **an array of data points**, not each data point. Please read the two examples in https://deck.gl/docs/api-reference/aggregation-layers/hexagon-layer#elevationaggregation 
- The DeckGL and ReactMapGL cameras are synchronized. You are trying to set them independently from each other, which will not work. It is strongly recommended that you use `StaticMap` instead of the fully interactive component from react-map-gl. See [get-started example](https://github.com/visgl/deck.gl/blob/master/examples/get-started/react/mapbox/app.js).
- To rotate the camera, set transition properties on DeckGL. https://deck.gl/docs/developer-guide/view-state-transitions
- As the term "transition" suggests, it goes from one state to another. You have to first set the view state to one bearing, and them update the view state to a new bearing to trigger the rotation. See [example](https://codesandbox.io/s/deckgl-camera-rotation-xp0kd?file=/src/App.js)
- Do not set the [around](https://deck.gl/docs/api-reference/core/linear-interpolator) parameter of `LinearInterpolator`. It should be in screen pixels, not lnglat.

**Reply by jaballogian** [reply]
Great. Thank you for the explanation. I appreciate it a lot. 👏🏻
